File: Python/11_container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        square_max = 0
        left = 0
        right = len(height)-1
        while left < right:
            width = right - left
            square_max = max(square_max, min(height[left], height[right]) * width )
            if height[left] <= height[right]:
                left += 1
            else:
                right -=1
        return square_max
        
       

File: Java/11_container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0, right = height.length-1;
        int maxArea = 0;
        while(left< right){
            int area = (right - left) * Math.min(height[right], height[left]);
            maxArea = Math.max(maxArea , area);
            if(height[left]< height[right]){
                left++;
            }else{
                right--;
            }
        }
        return maxArea;
        
    }
}